hermeto/core/package_managers/cargo/main.py

```diff
@@ -202,9 +216,26 @@ def _swap_sources_directory_for_subsitution_slot(template: str) -> dict:
     return toml_template
 
 
+def _old_config_is_present_in(package_dir: RootedPath) -> bool:
+    return (package_dir.path / ".cargo/config").exists()
+
+
 def _use_vendored_sources(package_dir: RootedPath, config_template: dict) -> ProjectFile:
     """Make sure cargo will use the vendored sources when building the project."""
-    cargo_config = package_dir.join_within_root(".cargo/config.toml")
+    # Cargo could be told to use vendored sources instead of a registry via .cargo/config.toml.
+    # Prior to cargo v1.39.0 .cargo/config.toml was known as .cargo/config.
+    # After v1.39.0 this name was considered obsolete, however .cargo/config would
+    # take precedence on .cargo/config.toml if present and the latter one would be ignored.
+    # The recommended practice for dealing with a situation when an older build system
+    # has to build a more modern project is to symlink .cargo/config.toml to .cargo/config.
+    # And vice versa: renaming .cargo/config to .cargo/config.toml would have no effect on
+    # any post-2019 toolchain.
+    # Refer to https://doc.rust-lang.org/cargo/reference/config.html for further details.
+    # Since we could potentially end up building a somewhat stale Rust-based
+    # Python extension it is better to check if there is an old-style config present and
+    # process it if found.
+    cf_name = ".cargo/config" if _old_config_is_present_in(package_dir) else ".cargo/config.toml"
```

Comment:
nitpick (readability): I still don't know how defining an extra function just to be able to do ^this is better than plain
```
cf_name = ".cargo/config.toml"
if (package_dir.path / ".cargo/config").exists():
  cf_name = ".cargo/config"
```
I think we should limit the one-liner syntax to elementary checks (possibly the named lambdas we already have elsewhere), but overall we should not abuse the syntax so as to define alias functions for simple checks, it's IMO easier to decode the check than to jump into the definition and back in the editor. My 2c.